test/module/irohad/ametsuchi/ametsuchi_test.cpp

```diff
@@ -853,3 +853,90 @@ TEST_F(AmetsuchiTest, TestingStorageWhenDropAll) {
   ASSERT_EQ(0, wsv->getPeers().value().size());
   new_storage->dropStorage();
 }
+
+TEST_F(AmetsuchiTest, FindTxByHashTest) {
+  auto storage =
+      StorageImpl::create(block_store_path, redishost_, redisport_, pgopt_);
+  ASSERT_TRUE(storage);
+  auto blocks = storage->getBlockQuery();
+
+  iroha::pubkey_t pubkey1, pubkey2;
+  pubkey1.at(0) = 1;
+  pubkey2.at(0) = 2;
+
+  CreateRole createRole;
+  createRole.role_name = "user";
+  createRole.permissions = {can_add_peer, can_create_asset, can_get_my_account};
+
+  Transaction tx1;
+  tx1.creator_account_id = "admin1";
+
+  tx1.commands.push_back(std::make_shared<CreateRole>(createRole));
+  CreateDomain createDomain;
+  createDomain.domain_id = "domain";
+  createDomain.user_default_role = "user";
+  tx1.commands.push_back(std::make_shared<CreateDomain>(createDomain));
+
+  CreateAccount createAccount;
+  createAccount.account_name = "user1";
+  createAccount.domain_id = "domain";
+  createAccount.pubkey = pubkey1;
+  tx1.commands.push_back(std::make_shared<CreateAccount>(createAccount));
+
+  CreateRole createRole2;
+  createRole2.role_name = "user2";
+  createRole2.permissions = {
+      can_add_peer, can_create_asset, can_get_my_account};
+
+  Transaction tx2;
+  tx2.commands.push_back(std::make_shared<CreateRole>(createRole2));
+  CreateDomain createDomain2;
+  createDomain2.domain_id = "domain2";
+  createDomain2.user_default_role = "user";
+  tx2.commands.push_back(std::make_shared<CreateDomain>(createDomain2));
+
+  Block block;
+  block.transactions.push_back(tx1);
+  block.transactions.push_back(tx2);
+  block.height = 1;
+  block.prev_hash.fill(0);
+  block.txs_number = block.transactions.size();
+  block.hash = iroha::hash(block);
+
+  {
+    auto ms = storage->createMutableStorage();
+    ms->apply(block, [](const auto &blk, auto &query, const auto &top_hash) {
+      return true;
+    });
+    storage->commit(std::move(ms));
+  }
+
+  auto tx1hash = iroha::hash(tx1).to_string();
+  auto tx2hash = iroha::hash(tx2).to_string();
+
+  auto numberOfCalls = 0;
+
+  blocks->getTxByHash(tx1hash).subscribe([tx1hash, &numberOfCalls](auto tx) {
+    ++numberOfCalls;
+    EXPECT_EQ(iroha::hash(tx).to_string(), tx1hash);
+  });
+
+  blocks->getTxByHash(tx2hash).subscribe([tx2hash, &numberOfCalls](auto tx) {
+    ++numberOfCalls;
+    EXPECT_EQ(iroha::hash(tx).to_string(), tx2hash);
+  });
+
+  auto tx3hash = "some garbage";
```

Comment:
As I understand this looks like another test. Better to separate them.

Reply:
Well, I agree with you from formal point of view but I think these two checks are very similar to each other. Another reason to let them live together is a huge amount of boilerplate code in each test case which is hard to eliminate without redesigning of whole ametsuchi test.

Reply:
Ok, just put the formal remark those tests may be separated in the future, when we will rework ametsuchi test.